sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -295,6 +295,41 @@ case class AlterViewAsCommand(
   }
 }
 
+/**
+ * A command for users to get views in the given database.
+ * If a databaseName is not given, the current database will be used.
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW VIEWS [(IN|FROM) database_name] [[LIKE] 'identifier_with_wildcards'];
+ * }}}
+ */
+case class ShowViewsCommand(
+    databaseName: Option[String],
+    tableIdentifierPattern: Option[String]) extends RunnableCommand {
+
+  // The result of SHOW VIEWS has three basic columns: namespace, viewName and isTemporary.
+  override val output: Seq[Attribute] = Seq(
+    AttributeReference("namespace", StringType, nullable = false)(),
+    AttributeReference("viewName", StringType, nullable = false)(),
+    AttributeReference("isTemporary", BooleanType, nullable = false)())
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    val db = databaseName.getOrElse(catalog.getCurrentDatabase)
+
+    // Show the information of views.
+    val views = tableIdentifierPattern.map(catalog.listViews(db, _))
+      .getOrElse(catalog.listViews(db, "*"))
+    views.map { tableIdent =>
+      val namespace = tableIdent.database.getOrElse("")
```

Comment:
How about `tableIdent.database.toSeq.quoted`? We should quote it if the database name has dot, to be consistent with the future v2 command view name.

Reply:
I see, updated in 6a4348ed9e289c6ca3cc557f9a575e1829f53fc0.

Reply:
Hmm, is this really updated?

Reply:
Oh..I just force pushed from another machine ... Sorry. will updated it back.

Reply:
Find it back in 7bd9cc662528a7fcfdb1f07b6565a0e6cac8512a. Sorry for the mistake.

Reply:
Emm, I'll change the `toArray` back to `toSeq` when I get back to my well configured dev machine...

Reply:
The `MultipartIdentifierHelper`(toSeq) will check empty, so the local temp view with empty database will got AnalysisException. Thus I'm reverting back to use `NamespaceHelper`(toArray) to make the code concise.